### PR TITLE
Add section on formatting to `Report`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

We did not mention, how it's possible to format a `Report`, this PR adds a small section on that topic.

## 🔗 Related links

- Closes https://github.com/hashintel/hash/issues/1595

## 🎥  Demo

image

image

### CODE REVIEW

File: packages/libs/error-stack/src/report.rs [hunk 1]
```diff
@@ -26,6 +26,17 @@ use crate::{
 /// Attachments, and objects [`provide`]d by a [`Context`], are directly retrievable by calling
 /// [`request_ref()`] or [`request_value()`].
 ///
+/// ## Formatting
+///
+/// `Report` implements [`Display`] and [`Debug`]. When utilizing the [`Display`] implementation,
```
Comment (Alfred-Mountfield) [inline packages/libs/error-stack/src/report.rs:31]: Can we add an example for each? 

Maybe it'd be better to restructure it into 3 subheadings for each type of formatting

Comment (TimDiekmann) [reply packages/libs/error-stack/src/report.rs:31]: Added the different outputs to the already provided example: https://github.com/hashintel/hash/pull/1597/commits/68d12d8c7e581355e6f7642e9711fa2eb1e84ad7

**Review by Alfred-Mountfield — DISMISSED** [review]
Great, thanks!

**Review by Alfred-Mountfield — APPROVED** [review]

**Comment by FlixCoder** [comment]
Thanks! That was fast, great! :D

**Pull request merged**